### PR TITLE
feat(coingecko): add endpoint for token market chart

### LINKED ISSUE #25: Add a function for /coins/{id}/market_chart coingecko endpoint
For more details look at #4

### PR DESCRIPTION
**Pull request by codeproline**
Hey there,

This PR adds endpoint for token market chart at `/coins/{id}/contract/{contract_address}/market_chart`.

Closes #25

### CODE REVIEW

File: packages/coingecko/src/query/routes/tokenInfo.ts [hunk 1]
```diff
@@ -195,7 +195,7 @@ function createMarketData(json: JSON.Obj): TokenInfoMarketData {
     price_change_percentage_60d: getStringProperty(json, "price_change_percentage_60d"),
     price_change_percentage_200d: getStringProperty(json, "price_change_percentage_200d"),
     price_change_percentage_1y: getStringProperty(json, "price_change_percentage_1y"),
-    market_cap_change_24h: getIntegerProperty<i32>(json, "market_cap_change_24h"),
+    market_cap_change_24h: getStringProperty(json, "market_cap_change_24h"),
```
Comment (Niraj-Kamdar) [inline packages/coingecko/src/query/routes/tokenInfo.ts:198]: nice catch!

File: packages/coingecko/src/query/routes/tokenMarketChart.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    request: {
+      headers: [],
+      urlParams,
+      body: "",
+      responseType: HTTP_ResponseType.TEXT,
+    },
+  });
+
+  if (!response || response.status !== 200 || !response.body) {
+    throw Error(response.statusText);
```
Comment (Niraj-Kamdar) [inline packages/coingecko/src/query/routes/tokenMarketChart.ts:40]:
```suggest
    throw new Error(response.statusText);
```

**Review by Niraj-Kamdar — CHANGES_REQUESTED** [review]
Please add integration tests also so CI can verify it. Going forward we want to improve our test coverage and have atleast one e2e test for any new function.

File: packages/coingecko/src/query/routes/tokenInfo.ts (hunk 1 above)
Comment (codeproline) [reply packages/coingecko/src/query/routes/tokenInfo.ts:198]: Thanks for pointing this out👍🏻

File: packages/coingecko/src/query/routes/tokenMarketChart.ts (hunk 2 above)
Comment (codeproline) [reply packages/coingecko/src/query/routes/tokenMarketChart.ts]: Thanks for pointing this out👍🏻

**Comment by codeproline** [comment]
Hey @Niraj-Kamdar, I have added new integration test for this function.

**Review by Niraj-Kamdar — APPROVED** [review]
LGTM!

**Comment by rihp** [comment]
500$ as reward has been paid out to @codeproline  👏 
txn : https://polygonscan.com/tx/0x4602d767ea5d1b585009685e588d11a137324e1e0c66725e1ad7b35a6d4bdb40

**Pull request merged**